File: ApogeeVGC/dex-moves/HitEffect.cpp
#include "HitEffect.h"

HitEffect::HitEffect(std::unique_ptr<OnHitFunc> on_hit,
	std::unique_ptr<SparseBoostsTable> boosts,
	std::unique_ptr<std::string> status,
	std::unique_ptr<std::string> volatile_status,
	std::unique_ptr<std::string> side_condition,
	std::unique_ptr<std::string> slot_condition,
	std::unique_ptr<std::string> pseudo_weather,
	std::unique_ptr<std::string> terrain,
	std::unique_ptr<std::string> weather) :
	on_hit(std::move(on_hit)),
	boosts(std::move(boosts)),
	status(std::move(status)),
	volatile_status(std::move(volatile_status)),
	side_condition(std::move(side_condition)),
	slot_condition(std::move(slot_condition)),
	pseudo_weather(std::move(pseudo_weather)),
	terrain(std::move(terrain)),
	weather(std::move(weather))
{
}

File: ApogeeVGC/Sim/Moves/HitEffect.cs
using System.Text.Json.Serialization;
using ApogeeVGC.Sim.Conditions;
using ApogeeVGC.Sim.Events;
using ApogeeVGC.Sim.Stats;

namespace ApogeeVGC.Sim.Moves;

public record HitEffect
{
    [JsonIgnore]
    public ResultMoveHandler? OnHit { get; set; }
    public SparseBoostsTable? Boosts { get; set; }
    public ConditionId? Status { get; init; }
    public ConditionId? VolatileStatus { get; set; }
    public ConditionId? SideCondition { get; set; } // Settable for pledge combo side conditions
    public ConditionId? SlotCondition { get; init; }
    public ConditionId? PseudoWeather { get; init; }
    public ConditionId? Terrain { get; init; }
    public ConditionId? Weather { get; init; }
}